runtime/compiler/runtime/Trampoline.cpp

```diff
@@ -183,7 +177,7 @@ void ppcCreateMethodTrampoline(void *trampPtr, void *startPC, void *method)
 
             // Now, if highest bit is on we need to clear the sign extend bits on 64bit CPUs
             // ** POWER4 pref fix **
-            if ( (dispatcher & 0x80000000) && proc == TR_PPCgp)
+            if ( (dispatcher & 0x80000000) && (!customP4 || (customP4 && TR::comp()->target().cpu.is(OMR_PROCESSOR_PPC_GP))))
```

Comment:
The same as above re checking customP4 twice. Also, not sure the transformation is equivalent. If customP4 == false, proc = TR_DefaultPPCProcessor, which is not the same as TR_PPCgp. Which means the if() will not be true.

Reply:
Good eye. I think we need to get rid of the `!customP4` over here.

Reply:
changing to `if ( (dispatcher & 0x80000000) && ((customP4 && TR::comp()->target().cpu.is(OMR_PROCESSOR_PPC_GP)))`

Reply:
@zl-wang could you please take a quick look to see if the code behaves as expected?

Reply:
there was some code to be improved, but it is not in the scope of this PR.  equiv transformations look fine to me at a quick glimpse.